src/journal-remote/journal-gatewayd.c

```diff
@@ -172,7 +174,10 @@ static ssize_t request_reader_entries(
                         return MHD_CONTENT_READER_END_OF_STREAM;
 
                 if (m->n_skip < 0)
-                        r = sd_journal_previous_skip(m->journal, (uint64_t) -m->n_skip + 1);
+                        if (m->since_set || m->until_set)
+                                r = sd_journal_previous_skip(m->journal, (uint64_t) -m->n_skip);
+                        else
+                                r = sd_journal_previous_skip(m->journal, (uint64_t) -m->n_skip + 1);
```

Comment:
Could you add a brief comment about the off-by-one difference here ?

Reply:
Actually the off-by-one was an error and the tests I added where not working so it didn't catch the error. I fixed both points in the last commit.